src/focus-trap-react.js

```diff
@@ -100,40 +130,36 @@ class FocusTrap extends React.Component {
         this.focusTrap.updateContainerElements(this.props.containerElements);
       }
 
-      if (prevProps.active && !this.props.active) {
-        // NOTE: we never let the trap return the focus since we do that ourselves
-        this.focusTrap.deactivate({ returnFocus: false });
-        if (this.returnFocusOnDeactivate) {
-          this.returnFocus();
-        }
-        return; // un/pause does nothing on an inactive trap
-      }
+      const hasActivated = !prevProps.active && this.props.active;
+      const hasDeactivated = prevProps.active && !this.props.active;
+      const hasPaused = !prevProps.paused && this.props.paused;
+      const hasUnpaused = prevProps.paused && !this.props.paused;
 
-      if (!prevProps.active && this.props.active) {
+      if (hasActivated) {
         this.updatePreviousElement();
         this.focusTrap.activate();
       }
 
-      if (prevProps.paused && !this.props.paused) {
-        this.focusTrap.unpause();
-      } else if (!prevProps.paused && this.props.paused) {
+      if (hasDeactivated) {
+        this.deactivateTrap();
+        return; // un/pause does nothing on an inactive trap
+      }
+
+      if (hasPaused) {
```

Comment:
With the pause stuff, `this.focusTrap.activate()` has some asynchronous functionality now if someone has defined `checkCanActivate`.

Do you think that will that be problematic?

Reply:
Unpausing doesn't cause the trap to be re-activated by going through the activation sequence and calling onActivate, etc. Am I missing something?

Reply:
The unit tests are passing...

Do you think you can make a failing unit test for this functionality?

Reply:
I don't think so. With the tests passing, I think we're good.